### ISSUE TITLE
PreconditionViolationException: Cannot create Launcher without at least one TestEngine; consider adding an engine implementation JAR to the classpath

### ISSUE THREAD

**Issue by jturolla** [JetBrains/spek#127]
Basic android setup on: http://jetbrains.github.io/spek/docs/latest/#setting-up-android results in the following error.

I'm sorry if this is trivial, but I'm not familiar on how to include a TestEngine and it would be nice to include this information on the Android section of Spek getting started guide.

If this is solved I'd be willing to contribute to the docs. 

By the way, I'm executing this using both with Android Studio using Spek Plugin and with `./gradlew build` , and I get the same error. 

```
Oct 06, 2016 10:02:15 PM org.junit.platform.launcher.core.ServiceLoaderTestEngineRegistry loadTestEngines
INFO: Discovered TestEngines with IDs: []

org.junit.platform.commons.util.PreconditionViolationException: Cannot create Launcher without at least one TestEngine; consider adding an engine implementation JAR to the classpath

	at org.junit.platform.commons.util.Preconditions.condition(Preconditions.java:161)
	at org.junit.platform.launcher.core.DefaultLauncher.<init>(DefaultLauncher.java:52)
	at org.junit.platform.launcher.core.LauncherFactory.create(LauncherFactory.java:42)
	at org.junit.platform.runner.JUnitPlatform.<init>(JUnitPlatform.java:88)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:98)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

build.gradle (module app)

```
buildscript {
  repositories {
    jcenter()
    maven { url 'https://maven.fabric.io/public' }
  }

  dependencies {
    classpath 'io.fabric.tools:gradle:1.+'
  }
}

repositories {
  jcenter()
  maven { url 'https://maven.fabric.io/public' }
  maven { url 'https://oss.sonatype.org/content/repositories/snapshots/' }
}

apply plugin: 'com.android.application'
apply plugin: 'android-apt'
apply plugin: 'kotlin-android'
apply plugin: 'me.tatarka.retrolambda'
apply plugin: 'realm-android'


android {
  compileSdkVersion 24
  buildToolsVersion "24.0.2"

  compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
  }

  defaultConfig {
    minSdkVersion 15
    targetSdkVersion 24
    versionCode 30
    versionName "1.5.0"

    testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    multiDexEnabled true
  }

  signingConfigs {
    release {
      storeFile file(RELEASE_STORE_FILE)
      storePassword RELEASE_STORE_PASSWORD
      keyAlias RELEASE_KEY_ALIAS
      keyPassword RELEASE_KEY_PASSWORD
    }
  }

  productFlavors {
    production {
      applicationId "com.mycompany.myapp"
    }

    qa {
      applicationId "com.mycompany.myapp.qa"
      minSdkVersion 18
      targetSdkVersion 24
    }
  }

  buildTypes {
    release {
      signingConfig signingConfigs.release
      minifyEnabled false
      proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
      buildConfigField "boolean", "USE_CRASHLYTICS", "true"
    }

    debug {
      buildConfigField "boolean", "USE_CRASHLYTICS", "false"
    }
  }

  lintOptions {
    disable 'InvalidPackage' // butterKnife
  }

  packagingOptions {
    exclude 'META-INF/services/javax.annotation.processing.Processor'
    exclude 'LICENSE.txt'
  }

  useLibrary 'org.apache.http.legacy'

  sourceSets {
    main {
      res.srcDirs = [
        'src/main/res/layouts/activity_job',
        'src/main/res/layouts/activity_login',
        'src/main/res/layouts/activity_main',
        'src/main/res/layouts/activity_offer',
        'src/main/res/layouts/activity_tasker',
        'src/main/res/layouts',
        'src/main/res'
      ]
    }

    main.java.srcDirs += 'src/main/kotlin'
    test.java.srcDirs += 'src/main/kotlin'
  }
}

dependencies {
  compile fileTree(include: ['*.jar'], dir: 'libs')

  androidTestCompile 'com.android.support:support-annotations:24.2.0'
  androidTestCompile 'com.android.support.test:runner:0.5'
  androidTestCompile 'com.android.support.test:rules:0.5'
  androidTestCompile 'com.android.support.test.espresso:espresso-core:2.2.2'
  androidTestCompile 'com.squareup.okhttp3:mockwebserver:3.4.1'
  androidTestCompile 'com.squareup.okhttp3:okhttp:3.4.1'
  androidTestCompile 'com.squareup.okio:okio:1.9.0'

  testCompile 'org.jetbrains.spek:spek-api:1.0.89'
  testCompile 'org.junit.platform:junit-platform-runner:1.0.0-M2'
  testCompile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"

  compile 'com.jakewharton:butterknife:8.4.0'
  apt 'com.jakewharton:butterknife-compiler:8.4.0'

  compile 'com.android.support:appcompat-v7:24.2.0'
  compile 'com.android.support:design:24.2.0'
  compile 'com.android.support:recyclerview-v7:24.2.0'
  compile 'com.android.support:cardview-v7:24.2.0'
  compile 'com.android.support:multidex:1.0.1'

  compile 'com.google.android.gms:play-services-analytics:9.4.0'
  compile 'com.google.android.gms:play-services-gcm:9.4.0'
  compile 'com.google.android.gms:play-services-location:9.4.0'

  compile 'com.astuetz:pagerslidingtabstrip:1.0.1'
  compile 'org.apache.commons:commons-lang3:3.4'
  compile 'com.github.leonardoxh:android-masks:1.1'
  compile 'org.greenrobot:eventbus:3.0.0'
  compile 'br.com.caelum.stella:caelum-stella-core:2.1.2'

  compile 'com.squareup.retrofit2:retrofit:2.1.0'
  compile 'com.squareup.retrofit2:converter-gson:2.1.0'
  compile 'com.squareup.okhttp3:okhttp:3.4.1'
  compile 'com.squareup.okhttp3:logging-interceptor:3.4.1'
  compile 'com.squareup.picasso:picasso:2.5.2'
  compile 'com.makeramen:roundedimageview:2.2.1'

  compile 'com.birbit:android-priority-jobqueue:2.0.0'
  compile 'com.helpshift:android-helpshift-aar:4.6.0'
  compile 'com.timehop.stickyheadersrecyclerview:library:0.4.3@aar'
  compile 'me.zhanghai.android.materialprogressbar:library:1.1.5'
  compile 'joda-time:joda-time:2.9.3'
  compile 'io.nlopez.smartlocation:library:3.2.7'

  compile('com.crashlytics.sdk.android:crashlytics:2.6.1@aar') {
    transitive = true;
  }

  compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
}

apply plugin: 'com.google.gms.google-services'
apply plugin: 'io.fabric'
```

build.gradle (project)

```
buildscript {
  ext.kotlin_version = '1.0.4'

  repositories {
    jcenter()
  }

  dependencies {
    classpath 'com.neenbedankt.gradle.plugins:android-apt:1.8'
    classpath 'com.android.tools.build:gradle:2.2.0'
    classpath 'com.google.gms:google-services:3.0.0'
    classpath 'me.tatarka:gradle-retrolambda:3.3.0-beta4'
    classpath 'io.realm:realm-gradle-plugin:1.2.0'
    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    classpath("org.junit.platform:junit-platform-gradle-plugin:1.0.0-M2")
  }
}

allprojects {
  repositories {
    jcenter()
  }
}
```

**Comment by raniejade** [comment]
Weird, `testCompile 'org.junit.platform:junit-platform-runner:1.0.0-M2'` should pull in the TestEngine. Is your project available online? If not can you indicate the structure of your gradle project.

Also, shouldn't this `test.java.srcDirs += 'src/main/kotlin'` be `test.java.srcDirs += 'src/test/kotlin'`?

**Comment by otomatik** [comment]
@jturolla I had the same error and fixed it by adding the following dependency:
```
testCompile 'org.jetbrains.spek:spek-junit-platform-engine:1.0.89'
````

**Comment by jturolla** [comment]
Nice, I'll test this out and get back with the result @otomatik

**Comment by raniejade** [comment]
@jturolla Did that fix your issue? Going to close this issue if yes :)

**Comment by jturolla** [comment]
@raniejade Yes, the issue is fixed.

**Comment by raniejade** [comment]
Fixed with #131 thanks @otomatik @jturolla

**Comment by FabianoLothor** [comment]
I solve it changing

`classpath "org.springframework.boot:spring-boot-gradle-plugin:2.1.8.RELEASE"`

to

`classpath "org.springframework.boot:spring-boot-gradle-plugin:2.2.0.RELEASE"`
